x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountsTokenStore.java

```diff
@@ -17,19 +25,37 @@
     /**
      * Verify the given token for encapsulated service account and credential
      */
-    boolean authenticate(ServiceAccountToken token);
+    void authenticate(ServiceAccountToken token, ActionListener<Boolean> listener);
```

Comment:
If the service token gets more sophisticated in the future, e.g. metadata etc, we may need to a listener that can take rich object. But for now, `Boolean` is sufficent.